Sources/ReadingListKit/PocketAccessLayerImplementation.swift

```diff
@@ -0,0 +1,145 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Apollo
+import MoSoCore
+
+public protocol ReadingListSession {
+    var token: String { get }
+    var guid: String { get }
+}
+
+protocol PocketAccessLayer {
+    var delegate: ReadingListModelDelegate? { get set }
+
+    func fetchSaves()
+    func initApolloClient()
+    func getItemForURL(_ urlString: String) async throws -> PocketGraph.ItemByURLQuery.Data.ItemByUrl
+    func archive(item: String)
+}
+
+public typealias ReadingListSessionProvider = () throws -> ReadingListSession
+
+enum PALError: Error {
+    case failedToFetchList(_ reason: String)
+    case failedToFetchItem(_ itemURLString: String)
+    case invalidAuthentication
+}
+
+enum PALState {
+    case idle
+    case fetching
+}
+
+class PocketAccessLayerImplementation: PocketAccessLayer {
+    let consumerKey: String
+    let pageSize: GraphQLNullable<Int> = GraphQLNullable<Int>(integerLiteral: 20)
+    var apolloClient: ApolloClient?
+    var pagination: PocketGraph.PaginationInput
+    var sessionProvider: ReadingListSessionProvider
+
+    private(set) var state: PALState = .idle
+
+    weak var delegate: ReadingListModelDelegate?
+
+    init(_ authTokenProvider: @escaping ReadingListSessionProvider, _ consumerKey: String) {
+        self.sessionProvider = authTokenProvider
+        self.consumerKey = consumerKey
+        self.pagination = PocketGraph.PaginationInput(after: .none, first: pageSize)
+    }
+
+    func initApolloClient() {
+        self.apolloClient = ApolloClient.createDefault(sessionProvider: sessionProvider, consumerKey: consumerKey)
+    }
+
+    // MARK: - Fetch Saves
+
+    func fetchSaves() {
+        if state == .fetching { return }
+        state = .fetching
+
+        let query = PocketGraph.FetchSavesQuery(
+            pagination: .some(pagination),
```

Comment:
This is a great suggestion, it fixes a few of my issues with this logic.

Reply:
Updated, the model now holds a String? that is the end of the last fetched page.
This will cause some knock-on changes to be needed in the Notification PR. But this is much better.